src/backend/main.ts

```diff
@@ -200,132 +205,151 @@ export async function updateOpenDataIndex() {
+  return Promise.all(
+    //
+    // Fetch data files
+    openDataFileNames.map(filename =>
+      s3GetJsonHelper({ Bucket: bucket, Key: `${filename}.json` }).then<[string, any]>(data => [filename, data]),
+    ),
+  ).then(entries => {
+    //
+    // Map data files into dictionary
+    const openDataIndex = entries.reduce((memo, [filename, data]) => {
+      memo[filename] = data.meta;
+      return memo;
+    }, {} as Record<string, any>);
+
+    //
+    // Produce open data index
+    // TODO: Model these (meta: OpenDataMeta, data: T = any)
+    const openData = {
+      meta: {
+        description: 'This is the open data site for the www.oiretutka.fi project',
+      },
+      data: openDataIndex,
     };
-    return acc;
-  }, {});
 
-  // Accumulate postal code level data into city level data
-  for (const postalCodeData of postalCodeLevelData) {
-    const city = postalCodeCityMappings.data[postalCodeData.postal_code];
+    //
+    // Push open data index to s3
+    return s3PutJsonHelper({ Bucket: bucket, Key: 'index.json', Body: openData });
+  });
+}
 
-    if (!city) {
-      console.warn(`mapPostalCodeLevelToCityLevelData: Skipping unknown postal code ${postalCodeData.postal_code}`);
-      continue;
+function mapPostalCodeLevelToCityLevelData(postalCodeLevelData: any[], bucket: string) {
+  return Promise.all([
+    //
+    // Fetch supporting data
+    s3GetJsonHelper({ Bucket: bucket, Key: 'postalcode_city_mappings.json' }),
+    s3GetJsonHelper({ Bucket: bucket, Key: 'population_per_city.json' }),
+  ]).then(([postalCodeCityMappings, populationPerCity]) => {
+    //
+    // Transform postal code level data to city level data
+
+    // Generate initial city-level aggregation set from the city population data
+    const resultsByCity = (populationPerCity.data as any[]).reduce((acc, data) => {
+      acc[data.city] = {
+        city: data.city,
+        population: data.population,
+        responses: 0,
+        fever_no: 0,
+        fever_slight: 0,
+        fever_high: 0,
+        cough_no: 0,
+        cough_mild: 0,
+        cough_intense: 0,
+        general_wellbeing_fine: 0,
+        general_wellbeing_impaired: 0,
+        general_wellbeing_bad: 0,
+        breathing_difficulties_no: 0,
+        breathing_difficulties_yes: 0,
+        muscle_pain_no: 0,
+        muscle_pain_yes: 0,
+        headache_no: 0,
+        headache_yes: 0,
+        sore_throat_no: 0,
+        sore_throat_yes: 0,
+        rhinitis_no: 0,
+        rhinitis_yes: 0,
+        stomach_issues_no: 0,
+        stomach_issues_yes: 0,
+        sensory_issues_no: 0,
+        sensory_issues_yes: 0,
+        longterm_medication_no: 0,
+        longterm_medication_yes: 0,
+        smoking_no: 0,
+        smoking_yes: 0,
+        corona_suspicion_no: 0,
+        corona_suspicion_yes: 0,
+      };
+      return acc;
+    }, {});
+
+    // Accumulate postal code level data into city level data
+    for (const postalCodeData of postalCodeLevelData) {
```

Comment:
Rest of the codebase seems to handle these kinds of operations with `.reduce`, but I opted to keep the `for` loop here.